src/workerd/api/global-scope.c++

```diff
@@ -248,6 +248,19 @@ kj::Promise<DeferredProxy<void>> ServiceWorkerGlobalScope::request(
     body = Body::ExtractedBody(jsStream.addRef());
   }
 
+  // If the request doesn't specify "Content-Length" or "Transfer-Encoding", set "Content-Length"
+  // to the body length if it's known. This ensures handlers for worker-to-worker requests can
+  // access known body lengths if they're set, without buffering bodies.
+  if (method != kj::HttpMethod::HEAD && method != kj::HttpMethod::GET &&
+      headers.get(kj::HttpHeaderId::CONTENT_LENGTH) == nullptr &&
+      headers.get(kj::HttpHeaderId::TRANSFER_ENCODING) == nullptr) {
+    KJ_IF_MAYBE(l, requestBody.tryGetLength()) {
```

Comment:
If the length isn't known, should we set `Transfer-Encoding: chunked` to emulate what an actual HTTP connection would do?